src/ol/format/kmlformat.js

```diff
@@ -323,6 +279,57 @@ ol.format.KML.ICON_ANCHOR_UNITS_MAP_ = {
 
 
 /**
+ * @param {Array.<ol.style.Style>|undefined} style Style.
+ * @param {string} styleUrl Style URL.
+ * @param {Array.<ol.style.Style>} defaultStyle Default style.
+ * @param {Object.<string, (Array.<ol.style.Style>|string)>} sharedStyles
+ * Shared styles.
+ * @return {ol.FeatureStyleFunction} Feature style function.
+ * @private
+ */
+ol.format.KML.createFeatureStyleFunction_ = function(
+    style, styleUrl, defaultStyle, sharedStyles) {
+  var findStyle = (
```

Comment:
It should not be necessary to create a new `findStyle()` function for every call to `createFeatureStyleFunction_()`. Instead, you could create a static function, e.g. `ol.format.KML.findStyle(styleUrl, defaultStyle, sharedStyles)`.